### DISCUSSION TITLE
Cards not rendering to DOM

### DISCUSSION THREAD

**Discussion by nashvegasnate** [nss-evening-cohort-14/evening-client-side#299] [Capstones!] [answered]
Name: Nathan Keith

Describe your issue: Can't seem to get list cards to render to DOM. It appears that the array is there within the component when I access the devTools, but they are not appearing.

I referenced the following discussion tickets:
#294

What I have Googled:
https://stackoverflow.com/questions/30465651/passing-keys-to-children-in-react-js
https://forum.freecodecamp.org/t/reactjs-my-card-component-is-not-rendering-why/388141/3

What I have tried:
- addressing routes
- created styled containers

Link to file/branch in GitHub: https://github.com/nashvegasnate/SetList/blob/nk-read/src/views/Lists.js

**Comment by TrinityChristiana** [comment]
Try checking your props.

**Comment by drteresavasquez** [comment]
I am also going to ask that you create a flow chat for data in your app. This will help you when you're passing props and working.

**Reply by nashvegasnate** [reply]
Attached is a link to the data flow chart that is now part of my README:
https://user-images.githubusercontent.com/76710923/121929491-e1662e00-cd06-11eb-8c4d-ea8b7cdd0811.png

**Comment by TrinityChristiana** [comment]
@nashvegasnate is updating their flow chart to show the following things:
1. Where each state is being instantiated (Where the `useState` hook is being used)
2. Where each state item (function and value) is being passed down as props
3. Where the set state function is being called and where the state value is being used.

**Reply by nashvegasnate** [reply]
See attached link to revised flow chart:
https://lucid.app/lucidchart/08d28a24-fc4b-4eb8-81ef-e7c5900d1891/edit?beaconFlowId=0C27D6294979F729&page=0_0#

**Comment by nashvegasnate** [comment, marked as answer]
Thanks to @tsekeres, I was able to get through the issue!
The main problem was due to my routes still being in a stage of nascence and not fully fleshed out, resulting in the failure of my components to load when the url changed when clicking on the link the view the cards. I had to add the component to the appropriate route so that the component would render when navigating to that page, i.e.:
```
<Route exact path='/lists' user={user} component={() => (
         <Lists user={user}/>)}
          />
```
It was a fairly simple fix; I just didn't set up my files in a proper, sequential order. Card are now rendering to the DOM! I think the video on dynamic routes was the only one I hadn't revisited, but now I have!

ALSO: I had to clean up my props. I had quite a few extraneous props being passed around, which could unnecessarily complicate things and looked horrible.
